tools/llvm2bpl/llvm2bpl.cpp

```diff
@@ -122,6 +123,7 @@ int main(int argc, char **argv) {
   pass_manager.add(new llvm::MergeArrayGEP());
   // pass_manager.add(new smack::SimplifyLibCalls());
   pass_manager.add(new llvm::Devirtualize());
+  pass_manager.add(new smack::MemoryAllocationChecker());
```

Comment:
This should not be enabled by default. You should introduce a command line option that enables it. See how this is done with `StaticUnroll` flag. Call this option e.g. `MemorySafety`. You should also modify the SMACK python script to include this option.